pubnub/test/integration/app_context/_utils.dart

```diff
@@ -0,0 +1,317 @@
+  }) {
+    return MembershipMetadataInput(
+      channelId,
+      custom: custom ??
+          {
+            'role': 'member',
+            'joined': DateTime.now().toIso8601String(),
+            'notifications': true,
+            'priority': 'normal',
+          },
+    );
+  }
+
+  /// Creates test channel member metadata input
+  static ChannelMemberMetadataInput createTestChannelMemberMetadata(
+    String uuid, {
+    Map<String, Object?>? custom,
+  }) {
+    return ChannelMemberMetadataInput(
+      uuid,
+      custom: custom ??
+          {
+            'role': 'member',
+            'invited_by': 'admin',
+            'permissions': 'read',
+            'status': 'active',
+          },
+    );
+  }
+
+  /// Cleans up test data by removing all metadata with test prefix
+  static Future<void> cleanupTestData(
+    PubNub pubnub,
+    String testPrefix,
+  ) async {
+    try {
+      // Clean up UUID metadata
+      final uuidsResult = await pubnub.objects.getAllUUIDMetadata(
+        filter: 'name LIKE "$testPrefix*"',
+        limit: 100,
+      );
+
+      if (uuidsResult.metadataList != null) {
+        for (final uuidMetadata in uuidsResult.metadataList!) {
+          try {
+            await pubnub.objects.removeUUIDMetadata(uuid: uuidMetadata.id);
+          } catch (e) {
+            // Ignore cleanup errors
+            print('Failed to cleanup UUID ${uuidMetadata.id}: $e');
+          }
+        }
+      }
+
+      // Clean up channel metadata
+      final channelsResult = await pubnub.objects.getAllChannelMetadata(
+        filter: 'name LIKE "$testPrefix*"',
+        limit: 100,
+      );
+
+      if (channelsResult.metadataList != null) {
+        for (final channelMetadata in channelsResult.metadataList!) {
+          try {
+            await pubnub.objects.removeChannelMetadata(channelMetadata.id);
+          } catch (e) {
+            // Ignore cleanup errors
+            print('Failed to cleanup channel ${channelMetadata.id}: $e');
+          }
+        }
+      }
+    } catch (e) {
+      // Ignore cleanup errors
+      print('Failed to cleanup test data: $e');
+    }
+  }
+
+  /// Waits for eventual consistency (used after operations that might need time to propagate)
+  static Future<void> waitForEventualConsistency([Duration? delay]) async {
+    await Future.delayed(delay ?? Duration(milliseconds: 500));
+  }
+
+  /// Creates multiple test UUIDs for bulk operations
+  static Future<List<String>> createMultipleTestUuids(
+    PubNub pubnub,
+    int count, {
+    String? namePrefix,
+  }) async {
+    final uuidIds = <String>[];
+
+    for (var i = 0; i < count; i++) {
+      final uuid = generateTestUuid('bulk-$i');
+      final metadata = createTestUuidMetadata(
+        name: '${namePrefix ?? 'Bulk Test User'} $i',
+      );
+
+      await pubnub.objects.setUUIDMetadata(metadata, uuid: uuid);
+      uuidIds.add(uuid);
+
+      // Small delay to avoid overwhelming the service
+      if (i % 10 == 9) {
```

Comment:
Are those delays required? Is there any consequence of not having them in place?